src/indexer/index_writer.rs

```diff
@@ -348,6 +348,13 @@ impl IndexWriter {
         self.operation_sender = sender;
     }
 
+    /// Accessor to the index
+    ///
+    /// The index is actually cloned.
+    pub fn index(&self) -> Index {
+        self.index.clone()
```

Comment:
Cloning Index is not super cheap.
Why not returning a reference? People could then clone it if required?

```suggestion
    pub fn index(&self) -> &Index {
        &self.index
```

Reply:
All right! Is there any reason not to do the same in https://github.com/tantivy-search/tantivy/blob/367f5da7827f5213b04efb8409f5ecc8374a9786/src/core/index.rs#L474?

Reply:
yes :)
One big difference is that Schema is super cheap to copy.